repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/scanner/ProjectScanner.java

```diff
@@ -311,35 +311,30 @@ private boolean testBuild(Build build, boolean targetFailing) {
     }
 
     private boolean thereIsDiffOnJavaFile(Build build, Build previousBuild) {
-        try {
-            GitHub gh = GitHubBuilder.fromEnvironment().build();
-
-            GHRateLimit rateLimit = gh.getRateLimit();
-            SimpleDateFormat dateFormat = new SimpleDateFormat("HH:mm:ss");
-            this.logger.debug("GitHub ratelimit: Limit: " + rateLimit.limit + " Remaining: " + rateLimit.remaining + " Reset hour: " + dateFormat.format(rateLimit.reset));
-
-            if (rateLimit.remaining > 2) {
-                GHRepository ghRepo = gh.getRepository(build.getRepository().getSlug());
-                GHCommit buildCommit = ghRepo.getCommit(build.getCommit().getSha());
-                GHCommit previousBuildCommit = ghRepo.getCommit(previousBuild.getCommit().getSha());
-                GHCompare compare = ghRepo.getCompare(previousBuildCommit, buildCommit);
-                GHCommit.File[] modifiedFiles = compare.getFiles();
-                for (GHCommit.File file : modifiedFiles) {
-                    if (file.getFileName().endsWith(".java")) {
-                        this.logger.debug("First java file found: " + file.getFileName());
-                        return true;
-                    }
-                }
-            } else {
-                this.logger.warn("You reach your rate limit for github, you have to wait " + rateLimit.reset + " to get datas. PRInformation will be null for build "+build.getId());
+        GHCompare compare = this.getCompare(build, previousBuild);
```

Comment:
`getCompare` can return null you have to check the value of compare to avoid a NPE here